src/Factory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Part of the Joomla Framework Registry Package
+ *
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Registry;
+
+/**
+ * Factory class to fetch Registry objects
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Factory
+{
+	/**
+	 * Format instances container - for backward compatibility with AbstractRegistryFormat::getInstance().
+	 *
+	 * @var    array
+	 * @since  __DEPLOY_VERSION__
+	 * @deprecated  2.0  Object caching will no longer be supported
+	 */
+	protected static $formatInstances = array();
+
+	/**
+	 * Returns a AbstractRegistryFormat object, only creating it if it doesn't already exist.
+	 *
+	 * @param   string  $type     The format to load
+	 * @param   array   $options  Additional options to configure the object
+	 *
+	 * @return  AbstractRegistryFormat  Registry format handler
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  \InvalidArgumentException
+	 */
+	public static function getFormat($type, array $options = array())
+	{
+		// Sanitize format type.
+		$type = strtolower(preg_replace('/[^A-Z0-9_]/i', '', $type));
+
+		/*
+		 * Only instantiate the object if it doesn't already exist.
+		 * @deprecated 2.0 Object caching will no longer be supported, a new instance will be returned every time
+		 */
+		if (!isset(self::$formatInstances[$type]))
+		{
+			$localNamespace = __NAMESPACE__ . '\\Format';
+			$namespace      = isset($options['format_namespace']) ? $options['format_namespace'] : $localNamespace;
+			$class          = $namespace . '\\' . ucfirst($type);
+
+			if (!class_exists($class))
+			{
+				// Were we given a custom namespace?  If not, there's nothing else we can do
+				if ($namespace === $localNamespace)
+				{
+					throw new \InvalidArgumentException(sprintf('Unable to load format class for type "%s".', $type), 500);
+				}
+
+				$class = $localNamespace . '\\' . ucfirst($type);
+
+				if (!class_exists($class))
+				{
+					throw new \InvalidArgumentException(sprintf('Unable to load format class for type "%s".', $type), 500);
```

Comment:
Neither exception is throwing a message with a class name.  Just saying that a class for the requested type can't be found.  Actually this is an improvement over the current message (which equates to basically "not found").